src/module.ts

```diff
@@ -66,24 +69,21 @@ export const plugin = new PanelPlugin<PanelOptions>(PlotlyPanel)
         defaultValue: '',
         category: ['X Axis'],
       })
-      .addCustomEditor({
+      .addMultiSelect<string, SelectFieldConfigSettings<string>>({
         path: 'yAxis.fields',
-        editor: MultiSelectValueEditor as any,
+        name: 'Field',
         settings: {
           options: [],
-          getOptions: getFieldOptions,
+          getOptions: (context) => getFieldOptions(context, (context.options as PanelOptions)?.yAxis.fields),
```

Comment:
It looks like this could result in assigning `undefined` or an array to this value, is undefined a valid value?

Reply:
Same question for other instances of this.

Reply:
The type declarations indicate that this is an optional value, so undefined should be fine. Also, in my experience the Grafana UI components handle undefined values well.